compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -185,20 +185,14 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     hir::AsyncGeneratorKind::Block,
                     |this| this.with_new_scopes(|this| this.lower_block_expr(block)),
                 ),
-                ExprKind::Await(expr) => {
-                    let dot_await_span = if expr.span.hi() < e.span.hi() {
-                        let span_with_whitespace = self
-                            .tcx
-                            .sess
-                            .source_map()
-                            .span_extend_while(expr.span, char::is_whitespace)
-                            .unwrap_or(expr.span);
-                        span_with_whitespace.shrink_to_hi().with_hi(e.span.hi())
+                ExprKind::Await(expr, await_kw_span) => {
+                    let await_kw_span = if expr.span.hi() < await_kw_span.hi() {
```

Comment:
How does this check work when macros are involved.

Reply:
It probably doesn't, but given this comparison was pre-existing, it probably didn't work before either. I'll try to create a test for it and fix it.